android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonFormActivity.java

```diff
@@ -540,7 +540,24 @@ public JSONObject getmJSONObject() {
 
     @Override
     public void setmJSONObject(JSONObject mJSONObject) {
-        this.mJSONObject = mJSONObject;
+        super.setmJSONObject(mJSONObject);
+    }
+
+    @Override
+    protected void initiateFormUpdate(JSONObject json) {
+        if (getForm() != null && ((getForm().getHiddenFields() != null && !getForm().getHiddenFields().isEmpty()) || (getForm().getDisabledFields() != null && !getForm().getDisabledFields().isEmpty()))) {
+            String strCount = json.optString(JsonFormConstants.COUNT);
+            if (StringUtils.isNotBlank(strCount)) {
+                int count = Integer.parseInt(strCount);
+                for (int i = 1; i <= count; i++) {
+                    JSONObject ithStepObject = json.optJSONObject(JsonFormConstants.STEP + i);
+                    JSONArray fieldsJsonObject = ithStepObject.optJSONArray(JsonFormConstants.FIELDS);
```

Comment:
Consider using this `com.vijay.jsonwizard.utils.FormUtils#getMultiStepFormFields` method to get all the form fields.